src/main/scala/org/constellation/domain/redownload/RedownloadService.scala

```diff
@@ -99,20 +102,21 @@ class RedownloadService[F[_]: NonEmptyParallel](
       (updated, ())
     }
 
-  def persistPeerProposal(peer: Id, height: Long, hash: String, reputation: Reputation): F[Unit] =
-    logger.debug(s"Persisting proposal of ${peer.short} at height $height and hash $hash") >> peersProposals.modify {
+  def persistPeerProposal(peer: Id, proposal: Signed[SnapshotProposal]): F[Unit] = {
+    logger.debug(s"Persisting proposal of ${peer.short} at height ${proposal.value.height} and hash ${proposal.value.hash}") >> peersProposals.modify {
       m =>
         val existing = m.getOrElse(peer, Map.empty)
         val updated =
-          if (existing.contains(height)) m
-          else m.updated(peer, existing + (height -> PersistedSnapshotProposal(hash, reputation)))
+          if (existing.contains(proposal.value.height)) m
+          else m.updated(peer, existing + (proposal.value.height -> proposal))
         val ignored = InvertedMap(updated.mapValues(a => takeHighestUntilKey(a, getRemovalPoint(maxHeight(a)))))
         (ignored, ())
     }
+  }
 
   def persistCreatedSnapshot(height: Long, hash: String, reputation: Reputation): F[Unit] =
     createdSnapshots.modify { m =>
-      val updated = if (m.contains(height)) m else m.updated(height, PersistedSnapshotProposal(hash, reputation))
+      val updated = if (m.contains(height)) m else m.updated(height, Signed(SignHelp.hashSign(hash, keyPair), SnapshotProposal(hash, height, reputation)))
```

Comment:
Same here, you should sign proposal, not snapshot

Reply:
Thanks for catching that